src/xb-builder-node.c

```diff
@@ -1219,3 +1222,26 @@ xb_builder_node_get_tokens (XbBuilderNode *self)
 	g_return_val_if_fail (self != NULL, NULL);
 	return priv->tokens;
 }
+
+/* private */
+void
+xb_builder_node_add_token_idx (XbBuilderNode *self, guint32 tail_idx)
+{
+	XbBuilderNodePrivate *priv = GET_PRIVATE (self);
+
+	g_return_if_fail (self != NULL);
+	g_return_if_fail (tail_idx != XB_SILO_UNSET);
+
+	if (priv->token_idxs == NULL)
+		priv->token_idxs = g_array_new (FALSE, FALSE, sizeof(guint32));
+	g_array_append_val (priv->token_idxs, tail_idx);
+}
+
+/* private */
+GArray *
+xb_builder_node_get_token_idxs (XbBuilderNode *self)
```

Comment:
Might help to add a comment by this function saying that its return value is nullable.

Reply:
Fixed, ta.